Add automatic boot document

bk201 opened the pull request:
**Problem:**
Users need a doc to use the new [automatic boot feature](https://github.com/rancher/harvester-installer/pull/33).
 
**Solution:**
The PR is the doc.

**Related Issue:**

https://github.com/rancher/harvester/issues/217


**Test plan:**
<!-- Make sure tests pass on the Circle CI. -->

bk201 committed:
Add automatic boot document

Signed-off-by: Kiefer Chang <kiefer.chang@suse.com>

innobead reviewed (changes requested):
LGTM, just a few comments to resolve.

bk201 committed:
ADD UEFI HTTP boot and ISC DHCPd config example

Signed-off-by: Kiefer Chang <kiefer.chang@suse.com>

bk201 committed:
Add credentials leak warning

Signed-off-by: Kiefer Chang <kiefer.chang@suse.com>

bk201 committed:
Remove kernel/initrd extraction section

- Add a link to iPXE example
- Make it clear the feature is available after version 0.2.0.

Signed-off-by: Kiefer Chang <kiefer.chang@suse.com>

gitlawr reviewed (approved):
LGTM

innobead reviewed (approved):
LGTM @bk201 well done!
